### PR TITLE
fix: handle invalid version validation for installation script

### PR DESCRIPTION
**Pull request by bhavyagosai**
fix: https://github.com/dojoengine/dojo/issues/1831

- dojoup script will fail with proper error handling when invalid version is mentioned by the user while installation.
- currently `downloading latest dojo` was being displayed in spite of the user downloading a mentioned version. Fixed that by displaying the message only when a version is not provided by the user in the command

### CODE REVIEW

File: dojoup/dojoup [hunk 1]
```diff
@@ -133,10 +136,21 @@ main() {
     RELEASE_URL="https://github.com/${DOJOUP_REPO}/releases/download/${DOJOUP_TAG}/"
     BIN_ARCHIVE_URL="${RELEASE_URL}dojo_${DOJOUP_VERSION}_${PLATFORM}_${ARCHITECTURE}.$EXT"
 
+    # Check if the version mentioned by user exists in the Dojo repository.
+    if ! curl --output /dev/null --silent --head --fail "$BIN_ARCHIVE_URL"; then
+      say "Version ${DOJOUP_VERSION} does not exist at ${RELEASE_URL%/}"
+      say "Please mention a valid version."
```
Comment (glihm) [inline dojoup/dojoup:142]: We may say to the user that if `-v` is not used, the latest stable version is automatically installed, wdyt?

File: dojoup/dojoup [hunk 2]
```diff
@@ -133,10 +136,21 @@ main() {
...
+      err "Aborting installation."
+    fi
+
     echo $BIN_ARCHIVE_URL
 
-    # Download and extract the binaries archive
+    # display message only if version is not mentioned by user
+    if [ ! -n "$DOJOUP_USER_VERSION" ]; then
```
Comment (glihm) [inline dojoup/dojoup:149]: Indentation missing for the line into the if.

**Review by glihm — COMMENTED** [review]
Great first pas @bhavyagosai!

Would you mind rebasing on main to have the latest fix for the error you have in the CI?

Only one minor comment, to ensure we give to the user a solution in case they are using `-v` without knowing the default behavior.

Also, we may propose a list of releases, but not sure of the pertinence here or if it adds some value as we're doing an `asdf` plugin.

File: dojoup/dojoup (hunk 2 above)
Comment (bhavyagosai) [reply dojoup/dojoup:149]: fixed! idk how I missed it haha!

File: dojoup/dojoup (hunk 1 above)
Comment (bhavyagosai) [reply dojoup/dojoup]: Changed it to this message: 

`Please specify a valid version, or omit -v to install the latest stable version automatically.`

File: dojoup/dojoup [hunk 3]
```diff
@@ -133,10 +136,21 @@ main() {
...
+    # Check if the version mentioned by user exists in the Dojo repository.
+    if ! curl --output /dev/null --silent --head --fail "$BIN_ARCHIVE_URL"; then
+      say "Version ${DOJOUP_VERSION} does not exist at ${RELEASE_URL%/}"
+      say "Please specify a valid version, or omit -v to install the latest stable version automatically."
+      err "Aborting installation."
+    fi
+
     echo $BIN_ARCHIVE_URL
 
+    # display message only if version is not mentioned by user
```
Comment (glihm) [inline dojoup/dojoup:148]:
```suggestion
    # Display message only if version is not mentioned by user.
```

File: dojoup/dojoup [hunk 4]
```diff
@@ -73,6 +73,9 @@ main() {
 
   DOJOUP_REPO=${DOJOUP_REPO-dojoengine/dojo}
 
+  # Store user specified version seperately
```
Comment (glihm) [inline dojoup/dojoup:76]:
```suggestion
  # Store user specified version seperately.
```

File: dojoup/dojoup [hunk 5]
```diff
@@ -133,10 +136,21 @@ main() {
     RELEASE_URL="https://github.com/${DOJOUP_REPO}/releases/download/${DOJOUP_TAG}/"
     BIN_ARCHIVE_URL="${RELEASE_URL}dojo_${DOJOUP_VERSION}_${PLATFORM}_${ARCHITECTURE}.$EXT"
 
+    # Check if the version mentioned by user exists in the Dojo repository.
+    if ! curl --output /dev/null --silent --head --fail "$BIN_ARCHIVE_URL"; then
+      say "Version ${DOJOUP_VERSION} does not exist at ${RELEASE_URL%/}"
```
Comment (glihm) [inline dojoup/dojoup:141]:
```suggestion
      say "Version ${DOJOUP_VERSION} does not match any release listed at https://github.com/dojoengine/dojo/releases."
```

**Review by glihm — APPROVED** [review]
Awesome, thank you for this fix @bhavyagosai and the comment in the issue regarding the pagination. 👍

Minor re-wording. :)

**Comment by bhavyagosai** [comment]
Thank you! Saw these rewordings. Will make sure to follow them from next time!

**Pull request merged**